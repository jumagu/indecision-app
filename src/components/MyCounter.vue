<script setup lang="ts">
// import { ref, computed } from 'vue';

// ? forma tradicional de hacerlo en JS
// const props = defineProps({
//   value: { type: Number, required: true },
// });

interface Props {
  value: number;
  text: string;
}

const props = defineProps<Props>();

// const counter = ref(props.value);
// const squareCounter = computed(() => counter.value * counter.value);

import { useCounter } from '../composables/useCounter';

const { counter, squareCounter } = useCounter(props.value);

const increment = () => {
  counter.value++;
};

const decrement = () => {
  counter.value--;
};
</script>

<template>
  <section>
    <h3>{{ props.text }}</h3>
    <h3>Counter: {{ counter }}</h3>
    <h3>Square: {{ squareCounter }}</h3>
  </section>

  <div>
    <button @click="decrement">-1</button>
    <button @click="increment">+1</button>
  </div>
</template>

<!-- scoped: estilos aplicados solo a este componente -->
<style scoped>
h1 {
  color: red;
}
</style>
