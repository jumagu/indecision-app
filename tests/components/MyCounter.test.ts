import { mount } from '@vue/test-utils';

import MyCounter from '@/components/MyCounter.vue';

describe('Tests in MyCounter component', () => {
  test('Should match snapshot', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: 5,
        text: 'Hello World',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Should render the counter value correctly', () => {
    // Arrange
    const value = 5;
    const text = 'Hello World';

    //  Act
    const wrapper = mount(MyCounter, {
      props: {
        value,
        text,
      },
    });

    // Assert
    expect(wrapper.find('h3').text()).toContain(text);
    expect(wrapper.find('[data-testid="counter-heading"]').text()).toContain(`Counter: ${value}`);
  });

  test('Should increment the counter value when +1 button is clicked', async () => {
    // Arrange
    const value = 5;
    const text = 'Hello World';

    //  Act
    const wrapper = mount(MyCounter, {
      props: {
        value,
        text,
      },
    });

    const [, btnIncrement] = wrapper.findAll('button');

    await btnIncrement.trigger('click');

    const counterText = wrapper.find('[data-testid="counter-heading"]').text();

    expect(counterText).toBe(`Counter: ${value + 1}`);
  });

  test('Should decrement the value in -2 when -2 button is clicked twice', async () => {
    // Arrange
    const value = 5;
    const text = 'Hello World';

    //  Act
    const wrapper = mount(MyCounter, {
      props: {
        value,
        text,
      },
    });

    const [btnDecrement] = wrapper.findAll('button');

    await btnDecrement.trigger('click');
    await btnDecrement.trigger('click');

    const counterText = wrapper.find('[data-testid="counter-heading"]').text();

    expect(counterText).toBe(`Counter: ${value - 2}`);
  });
});
