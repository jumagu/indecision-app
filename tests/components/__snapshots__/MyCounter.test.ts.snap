// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests in MyCounter component > Should match snapshot 1`] = `
"<section data-v-0352012b="">
  <h3 data-v-0352012b="">Hello World</h3>
  <h3 data-v-0352012b="" data-testid="counter-heading">Counter: 5</h3>
  <h3 data-v-0352012b="">Square: 5</h3>
</section>
<div data-v-0352012b=""><button data-v-0352012b="">-1</button><button data-v-0352012b="">+1</button></div>"
`;
